### PR TITLE
Add source info to TLS certificate assets and create them from host details

### PR DESCRIPTION
**Pull request by timopollmeier**
This PR contains various changes to allow gathering TLS certificates from scan data and a few other smaller improvements:
* The SHA-256 fingerprint has been added to the collected data and it is used to identify unique certificates and as the default name (instead of the MD5 fingerprint).
* All TLS certificate assets owned by a user must have unique SHA-256 fingerprints
* Information about the location (host and port) and origin (report or import) has been added
* TLS certificates are now collected from host details once scan of a host has finished
* It is no longer possible to overwrite the certificate data itself to ensure consistency
* Certificates can no longer be moved to the trashcan, only deleted
* The serial number has been added to the collected fields

**Checklist**:

- Tests N/A
- [x] [CHANGELOG](https://github.com/greenbone/gvmd/blob/master/CHANGELOG.md) Entry

### CODE REVIEW

File: src/schema_formats/XML/GMP.xml.in [hunk 1]
```diff
@@ -6091,7 +6096,7 @@ Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
     </ele>
     <ele>
       <name>copy</name>
-      <summary>The UUID of an existing TLS certificate</summary>
+      <summary>The UUID of an existing TLS certificate.</summary>
```
Comment (mattmundell) [inline src/schema_formats/XML/GMP.xml.in]: Summaries shouldn't have a fullstop.  The generator adds them.

File: src/manage.c [hunk 2]
```diff
@@ -623,6 +623,27 @@ parse_ssldetails (const char *ssldetails,
...
 
+/**
+ * @brief Get a string representation of a certificate format.
+ *
+ * @param[in]  certificate_format  The format as gnutls_x509_crt_fmt_t.
+ *
+ * @return A string representation of the format (e.g. "PEM" or "DER").
+ */
+const char*
+tls_certificate_format_str (gnutls_x509_crt_fmt_t certificate_format)
```
[the remark this answers is not in the document]
Comment (timopollmeier) [reply src/manage.c]: I've moved the function and `parse_ssldetails` in 11d20b71a5b372f780f32908fba5ee34e0ddfe85.

File: src/manage_migrators.c [hunk 3]
```diff
@@ -798,6 +798,339 @@ migrate_212_to_213 ()
...
+       "  activation_time, expiration_time,"
+       "  md5_fingerprint, sha256_fingerprint, serial, certificate_format)"
+       " SELECT make_uuid(), %llu,"
+       "        '%s', '%s', m_now(), m_now(),"
+       "        '%s', '%s', '%s', %d,"
+       "        %ld, %ld,"
+       "        '%s', '%s', '%s', '%s';",
+       owner,
+       sha256_fingerprint ? quoted_sha256_fingerprint : "",
+       "" /* comment */,
```
[the remark this answers is not in the document]
Comment (timopollmeier) [reply src/manage_migrators.c]: I've moved the comment so it's after the comma in 620ba4b7ca55170df59dbbfbd580f2ef69e916f2.

File: src/manage_migrators.c [hunk 4]
```diff
@@ -798,6 +798,339 @@ migrate_212_to_213 ()
...
+ *
+ * @return 0 success, -1 error.
+ */
+int
+migrate_213_to_214 ()
+{
+  iterator_t tls_certs;
+  char *previous_fpr = NULL;
+  user_t previous_owner = 0;
+  tls_certificate_t current_tls_certificate = 0;
```
[the remark this answers is not in the document]
Comment (timopollmeier) [reply src/manage_migrators.c]: I'm not sure if this exactly what you had in mind but I've separated the initializations from the declarations in 620ba4b7ca55170df59dbbfbd580f2ef69e916f2.

File: src/manage_migrators.c [hunk 5]
```diff
@@ -798,6 +798,339 @@ migrate_212_to_213 ()
...
+                 "       coalesce (report_hosts.start_time, reports.date)"
+                 " FROM report_host_details AS rhd"
+                 " JOIN report_hosts ON rhd.report_host = report_hosts.id"
+                 " JOIN reports ON report_hosts.report = reports.id"
+                 " WHERE source_description = 'SSL/TLS Certificate'"
+                 "    OR source_description = 'SSL Certificate'"
+                 " ORDER BY rhd.name, reports.owner, reports.id");
+
+  while (next (&tls_certs))
+    {
```
[the remark this answers is not in the document]
Comment (timopollmeier) [reply src/manage_migrators.c]: I've added comments explaining the iterators in 620ba4b7ca55170df59dbbfbd580f2ef69e916f2.

File: src/schema_formats/XML/GMP.xml.in (hunk 1 above)
Comment (timopollmeier) [reply src/schema_formats/XML/GMP.xml.in]: I've changed that in 1e59119cdd27e048e8788bba54db7170bbf1276f.
There are other places that have this, but I'd clean them up separately so the changes can be backported more easily.

File: src/manage_migrators.c (hunk 4 above)
[the remark this answers is not in the document]
Comment (mattmundell) [reply src/manage_migrators.c]: Yes that's what I meant, thanks.

File: src/manage_migrators.c (hunk 5 above)
[the remark this answers is not in the document]
Comment (mattmundell) [reply src/manage_migrators.c]: Thanks.

File: src/schema_formats/XML/GMP.xml.in (hunk 1 above)
Comment (mattmundell) [reply src/schema_formats/XML/GMP.xml.in]: Yes, good.

**Pull request merged**